components/base/kinematicbase/fake_kinematics.go

```diff
@@ -189,15 +239,58 @@ func WrapWithFakePTGKinematics(
 	return fk, nil
 }
 
+// WrapWithFakePTGKinematicsWithFrameReq is the request to WrapWithFakePTGKinematicsWithFrame.
+type WrapWithFakePTGKinematicsWithFrameReq struct {
```

Comment:
I'm not a fan of making another constructor.  Currently the other constructor is uncalled so we should just change the constructor to use the pattern we would rather be promoting.